### PR TITLE
- Added the concept of build statuses to projects

### PR DESCRIPTION
**Pull request by JonathanRosado**

### CODE REVIEW

File: controller/manager/ilm_builds.go [hunk 1]
```diff
@@ -97,18 +98,20 @@ func (m DefaultManager) CreateAllBuilds(projectId string, WsEmmitter *emitter.Em
 	go func(project *model.Project) {
 		sync := make(chan string)
 
+		projectStatus := model.BuildStatusFinishedSuccess
 		for _, test := range project.Tests {
 			m.CreateBuild(projectId, test.ID, model.NewBuildAction(model.BuildStartActionLabel), sync)
-			<-sync
+			if <-sync == model.BuildStatusFinishedFailed {
+				projectStatus = model.BuildStatusFinishedFailed
```
Comment (rqc) [inline controller/manager/ilm_builds.go]: wouldn't this `projectStatus` be overridden by the last build status incoming from the `sync` channel? I think it should be set the `model.BuildStatusFinishedLabel` when at least one of them has failed.

**Comment by rqc** [comment]
@JonathanRosado good job. Let's good over this code together and make sure that the project status is set in the correct moment.

File: controller/manager/ilm_builds.go (hunk 1 above)
Comment (JonathanRosado) [inline controller/manager/ilm_builds.go]: @rqc Isn't that what i'm doing? I'm setting `projectStatus` to `model.BuildStatusFinishedSuccess
` as its default value. Then, if a build fails, I'm setting `projectStatus` to `model.BuildStatusFinishedFailed`.
Technically `projectStatus` will be overwritten if another build fails (as stated in line 104), but this wouldn't change the semantics, as it is just re-setting it to `model.BuildStatusFinishedFailed`.

Comment (rqc) [inline controller/manager/ilm_builds.go]: ok!

**Pull request merged**